src/Compatibility/GenAPI/Microsoft.DotNet.GenAPI/IAssemblyVisitor.cs

```diff
@@ -0,0 +1,23 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.CodeAnalysis;
+
+namespace Microsoft.DotNet.GenAPI;
+
+public interface IAssemblyVisitor
```

Comment:
Can you document this, and mention its purpose?  Is it for testability - and if so, do you use that anywhere?  For that matter I also see that `IAssemblySymbolWriter` is an interface only ever used with a single type.  @ViktorHofer  do you think there is value in keeping these interfaces?

Reply:
Correct me if I'm wrong, but we use interfaces in case we one day decide to provide similar support to Visual Basic code, no?

Reply:
I'm not certain, but if we did add VB support we could add the interface at that time?  I tried to look back at the history and it was there from the start.  I couldn't see anywhere we actually made use of it.

Reply:
> I also see that IAssemblySymbolWriter is an interface only ever used with a single type. @ViktorHofer do you think there is value in keeping these interfaces?

Yes, the general idea for `IAssemblySymbolWriter` was to not depend on the specific implementation in the unit of code that depends on it (basically follow the Liskov Substitution Principle). Looks like GenAPIApp depends on the concrete type though so the interface doesn't serve much purpose today. There's probably room for improvement here and I would keep the interface.